src/Microsoft.AspNetCore.Server.Kestrel/Adapter/Internal/StreamSocketOutput.cs

```diff
@@ -42,18 +42,18 @@ public void Write(ArraySegment<byte> buffer, bool chunk)
                 }
 
                 var writableBuffer = _pipe.Writer.Alloc();
-
                 if (buffer.Count > 0)
                 {
                     if (chunk)
                     {
-                        ChunkWriter.WriteBeginChunkBytes(ref writableBuffer, buffer.Count);
-                        writableBuffer.WriteFast(buffer);
-                        ChunkWriter.WriteEndChunkBytes(ref writableBuffer);
+                        var writer = new WritableBufferWriter(writableBuffer);
+                        ChunkWriter.WriteBeginChunkBytes(ref writer, buffer.Count);
+                        writer.Write(buffer.Array, buffer.Offset, buffer.Count);
+                        ChunkWriter.WriteEndChunkBytes(ref writer);
                     }
                     else
                     {
-                        writableBuffer.WriteFast(buffer);
+                        writableBuffer.Write(buffer);
```

Comment:
Would be replaced by FastWrite implementation from https://github.com/dotnet/corefxlab/pull/1387